### ISSUE TITLE
Can't deserialize empty string.

### ISSUE THREAD

**Issue by Cykooz** [Pylons/colander#214]
Please, explain to me, why a node with an empty string value is considered as a missed node?
```python
>>> node = colander.SchemaNode(colander.String(), name='str_node')
>>> node.deserialize('')
...
Invalid: {'str_node': u'Required'}
```
Such behaviour affecting correct work of my code.
I use "Colander" in tests for validate the results returned by REST methods. Some fields in the results is required, but value of these fields can be an empty string.

In my case the result shown below is correct:
```python
{
    'name': 'Service name',
    'description': ''
}
```
But the next result must raise an exception:
```python
{
    'name': 'Service name'
}
```
How could I get this behaviour of Colander?

PS:
My question duplicate the closed issue #14, because it has not contains a full answer.

**Comment by leplatrem** [comment]
I have the exact same use case:  I could not figure out how to require the field presence while allowing empty strings!

**Comment by Natim** [comment]
It could be nice if a ``min_length=0`` could fix this problem.

**Comment by Cykooz** [comment]
I think the problem in this code:
```python
class String(SchemaType):
    ...
    def deserialize(self, node, cstruct):
        if not cstruct:
            return null
        ...
```

**Comment by leplatrem** [comment]
As specified in the tests, deserializing an empty string returns ``colander.null``
https://github.com/Pylons/colander/blob/1.0b1/colander/tests/test_colander.py#L1346-L1351

I'll submit a PR and see what comes out :)

**Comment by ametaireau** [comment]
Here is a link to a workaround: https://github.com/mozilla-services/readinglist/pull/253/files#diff-55f00e00a844509fa76123b7c7a6bba6

**Comment by alexsv** [comment]
any ideas if this will be in standard library?

**Comment by mmerickel** [comment]
Closed via #199.
